### PR TITLE
content compression fixed

### PR DESCRIPTION
**Pull request by markusstoll**
never worked properly

### CODE REVIEW

**Comment by prabier** [comment]
Hi Markus,

Why did you change log.debug with System.out.println? (last modification).

Philippe

**Comment by markusstoll** [comment]
tanks for bringing this up, this was changed just for testing purposes, I forgot to rollback this. fixed by now

File: Frameworks/Core/ERExtensions/Sources/er/extensions/appserver/ERXApplication.java [hunk 1]
```diff
@@ -2171,29 +2171,33 @@ public WOResponse dispatchRequestImmediately(WORequest request) {
...
 					else {
 						NSData input = response.content();
 						inputBytesLength = input.length();
-						compressedData = (inputBytesLength > 0) ? ERXCompressionUtilities.gzipByteArray(input._bytesNoCopy()) : null;
+						if(inputBytesLength > 0)
+						{
+							compressedData = ERXCompressionUtilities.gzipByteArrayAsNSData(input._bytesNoCopy(), 0, (int)inputBytesLength);
+						} else
+						{
+							compressedData = new NSData();
```
Comment (darkv) [inline Frameworks/Core/ERExtensions/Sources/er/extensions/appserver/ERXApplication.java]: You could use the constant NSData.EmptyData instead.

Comment (markusstoll) [inline Frameworks/Core/ERExtensions/Sources/er/extensions/appserver/ERXApplication.java]: i changed this to NSData.EmptyData

**Pull request merged**